src/Entity/Engine.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace Drupal\elastic_appsearch\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\elastic_appsearch\ElasticSearchInterface;
+use Drupal\elastic_appsearch\Utility\Database;
+use Drupal\elastic_appsearch\Utility\BatchHelper;
+
+/**
+ * Defines the Engine entity.
+ *
+ * @ConfigEntityType(
+ *   id = "elastic_appsearch_engine",
+ *   label = @Translation("Engine"),
```

Comment:
The label should be `@Translation("Elastic Appsearch Engine"),` because this is what display in UI.